schema_based_on_diagram.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE patients (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(100),
+    date_of_birth DATE
+);
+
+CREATE TABLE medical_histories (
+    id SERIAL PRIMARY KEY,
+    admited_at TIMESTAMP,
+    patient_id INTEGER REFERENCES patients(id) ON DELETE CASCADE,
+    status VARCHAR(100),
+    UNIQUE (patient_id)
+);
+
+CREATE TABLE invoices (
+    id SERIAL PRIMARY KEY,
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INTEGER REFERENCES medical_histories(id) ON DELETE CASCADE,
+    UNIQUE (medical_history_id)
+);
+
+
+CREATE TABLE treatments (
+    id SERIAL PRIMARY KEY,
+    type VARCHAR(150),
+    name VARCHAR(100)
+);
+
+CREATE TABLE medical_history_treatments (
+    medical_history_id INTEGER REFERENCES medical_histories(id) ON DELETE CASCADE,
+    treatment_id INTEGER REFERENCES treatments(id) ON DELETE CASCADE,
+    UNIQUE (medical_history_id, treatment_id)
+);
+
+CREATE TABLE invoice_items (
+    id SERIAL PRIMARY KEY,
+    unit_price DECIMAL,
+    quantity INTEGER,
+    total_price DECIMAL,
+    invoice_id INTEGER REFERENCES invoices(id) ON DELETE CASCADE,
+    treatment_id INTEGER REFERENCES treatments(id) ON DELETE CASCADE,
+    UNIQUE (invoice_id, treatment_id)
+);
+
```

Comment:
- [ ] As stated in the project requirement Kindly I recommend you to  add the FK indexes for all FOREIGN KEYS.
 
![Capture](https://user-images.githubusercontent.com/93598189/216135787-348b39d3-19bf-4fa9-885c-1f45c8703205.PNG)

Reply:
@Abenezer-Tilahun Hi, I'm not sure if I understand this requirement. Are we supposed to create indexes for the Fk ? For example, "CREATE INDEX idx_patient_id ON medical_histories (patient_id);" etc

Reply:
@sebalp100 Yeah Just Like the example you mentioned,